File: Data Structures & Algorithms/search-2d-matrix/submission-6.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {
        int ROWS = matrix.Length, COLS= matrix[0].Length;

        int left = 0, right = ROWS * COLS - 1;
        while (left <= right) {
            int middle = left + (right - left) / 2;
            int row = m / COLS, int col = m % COLS;

            if (target > matrix[row][col]) {
                left = middle + 1;
            } else if (target < matrix[row][col]) {
                right = middle - 1;
            } else {
                return true;
            }
        } 
        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        res = 0

        while left < right:
            currentSum = numbers[left] + numbers[right]

            if currentSum < target: 
                left += 1
            if currentSum > target:
                print(right)
                right -= 1
            if currentSum == target:
                return [left+1, right-1]
        return []
        